aws/rust-runtime/aws-config/src/imds/client/token.rs

```diff
@@ -133,7 +133,7 @@ impl TokenMiddleware {
             .call(operation)
             .await
             .map_err(ImdsError::FailedToLoadToken)?;
-        let expiry = response.expiry - TOKEN_REFRESH_BUFFER;
```

Comment:
Good catch! I forgot that the ExpiringCache was already considering the buffer time.